FrescoImageLoader/src/main/java/com/github/piasy/biv/loader/fresco/FrescoImageLoader.java

```diff
@@ -112,6 +116,20 @@ protected void onFail(final Throwable t) {
                     callback.onFail((Exception) t);
                 }
             }, mExecutorSupplier.forBackgroundTasks());
+
+            closeSource(parent);
+            saveSource(parent, source);
+        }
+    }
+
+    private void saveSource(BigImageView parent, DataSource target) {
+        mViewSourceMap.put(parent.hashCode(), target);
+    }
+
+    private void closeSource(BigImageView parent) {
+        DataSource source = mViewSourceMap.remove(parent.hashCode());
+        if (source != null) {
+            source.close();
```

Comment:
According to Fresco's doc, `close` should be enough.

Reply:
Okay, so no changes required here for Fresco?

I have pushed the other requested changes to ImageLoader interface.